src/rust/engine/options/src/config.rs

```diff
@@ -40,36 +123,80 @@ impl Config {
                 e
             )
         })?;
-        if !config.is_table() {
-            return Err(format!(
+
+        fn add_section_to_interpolation_map(
+            imap: &InterpolationMap,
+            section: Option<&Value>,
+        ) -> Result<InterpolationMap, String> {
```

Comment:
This is a slightly weird method signature. Slightly more idiomatic would be to take the `imap` by value, since the clone is guaranteed to happen. That way if a caller has an owned clone already, they avoid one clone.

Reply:
Yeah, good point. Will fix.